### PR TITLE
Fix deadlock and add better test

### PR DESCRIPTION
**Pull request by dmorawetz**

### CODE REVIEW

File: ikisocket.go [hunk 1]
```diff
@@ -78,7 +79,7 @@ type EventPayload struct {
 	// Unique connection UUID
 	SocketUUID string
 	// Optional websocket attributes
-	SocketAttributes map[string]string
+	SocketAttributes map[string]interface{}
```
Comment (antoniodipinto) [inline ikisocket.go:82]: I would prefer to leave it as string since if you use interface you're forced to cast every result

Comment (dmorawetz) [reply ikisocket.go:82]: True. What do you think of adding convenience methods to get certain types? Something like

```go
func (kws *Websocket) GetAttribute(key string) interface{}
func (kws *Websocket) GetStringAttribute(key string) string
func (kws *Websocket) GetIntAttribute(key string) int
...
func (kws *Websocket) SetAttribute(key string, value interface{})
```

Comment (antoniodipinto) [reply ikisocket.go:82]: Perfect approach, I think  int, string, inferface no more types supported. Maybe when GO will support generic we can rewrite this with generics

Comment (dmorawetz) [reply ikisocket.go:82]: Great, I will add it. I don't think that GO will ever get generics, though.

Comment (dmorawetz) [reply ikisocket.go:82]: Done

**Pull request merged**